docs/decisions/0025-abstract-workflow-pattern.rst

```diff
@@ -0,0 +1,46 @@
+0025 Adding an abstract workflow pattern
+****************************************
+
+Status
+======
+**In progress** (March 2025)
+
+Context
+=======
+We have decided to add a workflow pattern and related Django app (and abstract models)
+to our codebase. This will allow us to manage and persist workflow state in an
+auditable and customizable manner.
+The first concrete implementation of workflows we plan to represent
+is a provisioning workflow (see prior ADR for the HTTP API portion of this workflow) -
+a flow that allows clients to provision new Enterprise Customer, Catalog,
+and Subsidy records programmatically via a single API endpoint
+(or other types of single entrypoint interfaces). Notably, we’d like to provide
+an abstract implementation of workflows to support similar future concrete
+implementations - e.g. automated *onboarding* of customers.
+
+Alternatives Considered
```

Comment:
One possible alternative to add is celery tasks as composable workflow steps.  They are also composable, modular, introspectable, auditable, etc. so maybe speak to what makes them inadequate for this specific use case.